packages/experiments/src/utilities/keysequence/IKeySequence.ts

```diff
@@ -0,0 +1,87 @@
+import { ktpPrefix, ktpSeparator } from '../keytip/KeytipUtils';
+
+export interface IKeySequence {
+  keys: string[];
+}
+
+/**
+ * Tests for equality between two IKeySequences
+ *
+ * @param seq1 - First IKeySequence
+ * @param seq2 - Second IKeySequence
+ * @returns {boolean} T/F if the two sequences are equal
+ */
+export function keySequencesAreEqual(seq1: IKeySequence, seq2: IKeySequence): boolean {
+  let keyCodes1 = seq1.keys.join();
+  let keyCodes2 = seq2.keys.join();
+  return keyCodes1 === keyCodes2;
+}
+
+/**
+ * Tests for equality between two arrays of IKeySequence
+ *
+ * @param seq1 - First IKeySequence[]
+ * @param seq2 - Second IKeySequence[]
+ * @returns {boolean} T/F if the sequences are equal
+ */
+export function fullKeySequencesAreEqual(seq1: IKeySequence[], seq2: IKeySequence[]): boolean {
+  if (seq1.length !== seq2.length) {
+    return false;
+  }
+
+  for (let i = 0; i < seq1.length; i++) {
+    if (!keySequencesAreEqual(seq1[i], seq2[i])) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * Tests if 'seq' is present in 'sequences'
+ *
+ * @param sequences - Array of IKeySequence
+ * @param seq - IKeySequence to check for in 'sequences'
+ * @returns {boolean} T/F if 'sequences' contains 'seq'
+ */
+export function keySequencesContain(sequences: IKeySequence[], seq: IKeySequence): boolean {
+  for (let i = 0; i < sequences.length; i++) {
```

Comment:
Seems like this is just an implemetation of includes, unfortunately this doesn't work in IE11, but we can add an implementation in utilities (like there is for findIndex) to use to reduce boilerplate here in a few places. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
@christiango can we use includes() if we have an array of objects? (in this case, IKeySequence[]) I would think then we would need an object deep equals function...

at its base IKeySequence.keys is an array of strings, but that is an equality check we do with .join(), it wouldn't be an includes() check there

Reply:
You can use  [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) if you need more elaborate comparison logic